pkg/cmd/server/apis/config/types.go

```diff
@@ -871,6 +875,13 @@ type OAuthTemplates struct {
 	Error string
 }
 
+type ExternalOAuthConfig struct {
```

Comment:
@liggitt you ready to start an `AuthenticationConfig` or shall we wait until we split configs and just a little more ugly here?  I don't feel strongly.

Reply:
Do you want me to move this under MasterAuthConfig like for the WebToken stuff ?
Wasn't sure this was appropriate.

Reply:
I can have a simple ExternalOAuthMetadataFile string in there and avoid a new structure